encrypted-forms.rst

```diff
@@ -0,0 +1,227 @@
+
+1. Install OpenSSL (Windows only)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+For Windows, download and run the OpenSSL installer appropriate for your system from OpenSSL for Windows. When it asks whether to copy the DLLs to the Windows system directory or to the :file:`/bin` directory, choose the :file:`/bin` directory (either will work, but this will minimize the pollution of the Windows system directory)
+
+2. Constructing the RSA Key Pair
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ 
+If you are on Windows, open a powershell or command prompt window. Change directories to the /bin directory in the OpenSSL directory. e.g.,
+
+.. code-block:: doscon
+
+  > cd C:\OpenSSL-Win32\bin
+  
+If you are on a Mac, open the terminal. Change directories to your Desktop.
+
+.. code-block:: console
+
+  $ cd ~/Desktop
+   
+3. Create a private key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following command will create a 2048-bit private key and write it to the :file:`MyPrivateKey.pem` file. This may complain about a missing configuration file. You can ignore this warning.
+
+If you are on Windows, run:
+
+.. code-block:: doscon
+
+  > openssl genpkey -out MyPrivateKey.pem -outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048
+  
+.. warning::
+  
+  **On Powershell**
+  
+    Check **$env:path** to be sure :file:`path\\OpenSSL-Win64\\bin` is in there.If it is not, run the following command in Powershell:
+	
+  .. code-block:: console
+	
+    > $env:path = $env:path + ";path to OpenSSL-Win64\bin"
+	  
+If you are on a Mac, run:
+
+.. code-block:: console
+
+  $ openssl genrsa -out MyPrivateKey.pem 2048
+  
+4. Extract a public key
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, you need to extract the public key for this private key.
+
+Run the following command:
+
+.. code-block:: console
+
+  openssl rsa -in MyPrivateKey.pem -inform PEM -out MyPublicKey.pem -outform PEM -pubout
+  
+This may also complain about a missing configuration file. You can ignore this warning.
+
+5. Storing and using the keys
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Move the :file:`MyPrivateKey.pem` file to a secure location. It does not have a password encoding it, so anyone can decrypt your data if they have access to this file. This is the private key file that you will give to ODK Briefcase when decrypting the data.
+
+6. Updating the public_key field in the XLSForm settings worksheet.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Open the :file:`MyPublicKey.pem` file in Notepad (Windows) or TextEdit (Mac). Remove the leading and trailing '---...' lines. Then remove all new-lines (carriage-returns) from the file. Select the resulting very-long string, and paste it into the **public_key field** in the XLSForm settings worksheet. This very-long string will become the *base64RsaPublicKey* attribute in the resulting encrypted form definition.
```

Comment:
Since the user is presumably on a terminal/console, couldn't you just print the contents into the terminal and cut/paste from there? Also, a command that includes stripping the `---` and new-lines (what?!?) would be less error prone.

Also - there are *really* new lines that need to be removed? That is surprising. IF that is true, I would call it a bug or bad implementation.

Reply:
@adammichaelwood `MyPublicKey.pem` file looks like this

![screenshot 271](https://user-images.githubusercontent.com/21216969/30951383-4ee3af82-a440-11e7-80dd-b5948eac4418.png)

Well, we can actually use `less MyPublicKey.pem` to print the contents on the terminal but for now, the first line and the last line are required to be removed manually, as it signifies whether it is a private key or public key.I will correct the carriage-returns(new lines) part as it is not creating any problem

Reply:
@adammichaelwood should I add the command `less MyPublicKey.pem` as a tip or should I remove the notepad part and replace it with this?

Reply:
I would just tell people to open it, and to copy the part inside the `----BEGIN/END----` lines.

Then, I would add a "tip" mentioning notepad, textedit, and `less` as suggestions. 

Reason for this:
coders and other tech people know how to open text files without causing problems. So suggesting a particular application is only needed for non-tech people who might otherwise think it is a good idea to open it in like Word or Google Drive